modular_citadel/code/game/objects/items/balls.dm

```diff
@@ -0,0 +1,69 @@
+// Tennis Balls
+
+/obj/item/toy/tennis
+	name = "tennis ball"
+	desc = "A classical tennis ball. It appears to have faint bite marks scattered all over its surface."
+	icon = 'modular_citadel/icons/obj/balls.dmi'
+	icon_state = "tennis_classic"
+	item_icons = list(
+		slot_l_hand_str = 'modular_citadel/icons/mob/inhands/balls_left.dmi',
+		slot_r_hand_str = 'modular_citadel/icons/mob/inhands/balls_right.dmi',
+		slot_l_ear_str = 'modular_citadel/icons/mob/mouthball.dmi',
+		slot_r_ear_str = 'modular_citadel/icons/mob/mouthball.dmi',
+		slot_head_str = 'modular_citadel/icons/mob/mouthball.dmi'
+		)
+	item_state = "tennis_classic"
+	slot_flags = SLOT_HEAD | SLOT_EARS
+	throw_range = 14
+	w_class = ITEMSIZE_SMALL
+
+/obj/item/toy/tennis/rainbow
+	name = "pseudo-euclidean interdimensional tennis sphere"
+	desc = "A tennis ball from another plane of existence. Really groovy."
+	icon_state = "tennis_rainbow"
+	item_state = "tennis_rainbow"
+
+	var/list/squeak_sounds = list(
+		'modular_citadel/sound/items/toysqueak1.ogg',
+		'modular_citadel/sound/items/toysqueak2.ogg',
+		'modular_citadel/sound/items/toysqueak3.ogg'
+		)
+
+/obj/item/toy/tennis/rainbow/attack_self(mob/user as mob)
+	playsound(src.loc, pick(squeak_sounds), 50, 1,)
+
+/obj/item/toy/tennis/red
+	name = "red tennis ball"
+	desc = "A red tennis ball."
+	icon_state = "tennis_red"
```

Comment:
Make sure you can't accomplish the same results by simply setting the `color` var on a normal tennis ball. It would mean you'd need only one item sprite for the basic tennis balls, as well as one for the rainbow and the donator subtype.

Reply:
...I planned it as mere port with slight modifications :v

Reply:
The tennis balls have white stripes that'd be tinted by the color var. Furthermore, the shades of the various variants are hueshifted, so converting the sprites to greyscale would make them suck.

Reply:
That's fair, so the color var wouldn't work.